src/main/jhipster/src/main/webapp/app/shared/germplasm/service/germplasm.service.ts

```diff
@@ -19,6 +20,24 @@ export class GermplasmService {
             germplasmSearchRequest, { params: options, observe: 'response' });
     }
 
+    downloadGermplasmTemplate(updateFormat?: boolean): Observable<HttpResponse<Blob>> {
+        const url = SERVER_API_URL + `crops/${this.context.cropName}/germplasm/templates/xls?programUUID=` + this.context.programUUID + `&updateFormat=` + updateFormat;
+        return this.http.get(url, { observe: 'response', responseType: 'blob' });
+    }
+
+    importGermplasmUpdates(germplasmUpdates: any): Observable<HttpResponse<Germplasm[]>> {
+        return this.http.patch<any>(SERVER_API_URL + `crops/${this.context.cropName}/germplasm?programUUID=` + this.context.programUUID,
+            germplasmUpdates, { observe: 'response' });
+    }
+
+    getGermplasmNameTypes(codes: string[]): Observable<GermplasmNameTypeModel[]> {
+        return this.http.get<GermplasmNameTypeModel[]>(SERVER_API_URL + `crops/${this.context.cropName}/germplasm/name-types?codes=` + codes.join(','));
+    }
+
+    getGermplasmAttributes(codes: string[]): Observable<GermplasmAttributeModel[]> {
```

Comment:
Should this be in `attributes.service.ts`?

Reply:
I created the same method in IBP-4098 and I put it in this service too, given that it's tightly related to the germplasm, and also to mirror the api resource structure. I did reuse the `attribute.model.ts` though, and left a TODO to move it closer to the germplasm model (actually the current germplasm model could also be moved to the shared folder, and be renamed to germplasm-search-response.model, so that we can have a simplified germplasm model later, but all this can be done in the future...)